### ISSUE TITLE
[Json to Dart]Not generate double variable correctlly

### ISSUE THREAD

**Issue by plitto007** [quicktype/quicktype#1533]
If in json data, i have a value like this: ` "Distance": 20.0,`
The result will be` int distance;`

**Comment by Sun3** [comment]
Yes, I have the same results. I had to do something like this to make it work ```20.1234```

**Comment by azizainunnajib** [comment]
is there any update in this issue?

**Comment by schani** [comment]
I just retested this on current master with your example, an input containing `"Distance": 20.0`, and it now generates `double distance;` in the Dart output, not `int`. quicktype's JSON parser now preserves the decimal point when scanning numeric literals, so a value like 20.0 is correctly inferred as a double even though it has no significant fractional digits. You can check it yourself with `node dist/index.js --lang dart sample.json`. This fix will be included in the next release. If you can still reproduce the int inference on a current build, please reopen or comment with your sample and I'll take another look.
